### PR TITLE
feat: websocket endpoint

### PR DESCRIPTION
**Pull request by mathnogueira**
This PR adds an endpoint to handle Websocket connections

## Changes

- This only adds a `/ws` endpoint that does nothing yet;
- It creates the structure to handle message routing based on the message `type` attribute.

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Review by schoren — APPROVED** [review]
this looks awesome!

File: server/go/websocket/router.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	return &Router{
+		routes: make(map[string]MessageExecutor),
+	}
+}
+
+func (r *Router) Add(messageType string, executor MessageExecutor) {
+	r.routes[messageType] = executor
+}
+
+func (r *Router) ListenAndServe(addr string) {
```
Comment (xoscar) [inline server/go/websocket/router.go:28]: noob question, where does r come from here? Is it somehow getting injected?

File: server/go/websocket/router.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+type MessageExecutor func(*websocket.Conn, Message)
+
+type Router struct {
+	routes map[string]MessageExecutor
+}
+
+func NewRouter() *Router {
+	return &Router{
+		routes: make(map[string]MessageExecutor),
```
Comment (xoscar) [inline server/go/websocket/router.go:20]: Another noob one. What are the route names/message types that would be available here?

File: server/go/websocket/router.go (hunk 1 above)
Comment (schoren) [reply server/go/websocket/router.go:28]: The function signature means that the function is attached to the struct `Router`. That's the go way to create something similar to methods. You can think of the `r` as `this` (though not quite the same).  You can read [a more precise definition](https://go.dev/tour/methods/1)

File: server/go/websocket/router.go (hunk 2 above)
Comment (schoren) [reply server/go/websocket/router.go:20]: What that code means is that you will have a way to map a message type to an "object" that knows how to handle that specifc type

**Pull request merged**